hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java

```diff
@@ -1696,7 +1696,29 @@ public void visitRegionState(Result result, final RegionInfo regionInfo, final S
       if (localState.matches(State.OPEN, State.OPENING, State.CLOSING, State.SPLITTING,
         State.MERGING)) {
         assert regionLocation != null : "found null region location for " + regionNode;
-        regionStates.addRegionToServer(regionNode);
+
+        // If the startcode of the regionLocation expired, and neither TRSP on region nor SCP on
```

Comment:
How could this happen if the persistent region meta states are OPEN/OPENING but in the memory there are no TRSP/SCP? Is there some consistent issues of the region info? Can you explain more here? Thanks.

Reply:
eg. user bypass the proc or delete the pv-xxx.log

Reply:
For these cases, I would perfer let users use HBCK2 to fix the inconsistency, as these are all damages from outside the system.

Reply:
> For these cases, I would perfer let users use HBCK2 to fix the inconsistency, as these are all damages from outside the system.

IMO，HBCK2 is one way, but ability of self-healing is also useful, as it could reduce operation and maintenance difficulties, especially for novice users.

Reply:
For me, I prefer we do not do these 'smart' work, as you do not know how users will damage the system, so the 'self-healing' may become 'self-killing'...

Reply:
Ok, get it, thanks all the same for the reviewing.